paylike.php

```diff
@@ -109,225 +151,221 @@ public function renderForm()
+					$order = new Order((int)$id_order);
+					$order->setCurrentState((int)Configuration::get('PS_OS_REFUND'), $this->context->employee->id);
+
+					$id_cart = $refundrequest->transaction->custom->cartId;
+					$msg = new Message();
+					$message = strip_tags($message, '<br>');
+					if (Validate::isCleanHtml($message))
+					{
+						if (self::DEBUG_MODE)
+							PrestaShopLogger::addLog('PaymentModule::validateOrder - Message is about to be added', 1, null, 'Cart', (int)$id_cart, true);
+
+						$msg->message = $message;
+						$msg->id_cart = (int)$id_cart;
+						$msg->id_customer = (int)($order->id_customer);
+						$msg->id_order = (int)$order->id;
+						$msg->private = 1;
+						$msg->add();
+					}
 
-		return $output;
-}
+					$this->context->controller->confirmations[] = $this->l('Refunded successfully').'. '.$this->l('Refunded').' '.$refundrequest->transaction->currency.Tools::getValue('paylike_amount_to_refund');
+				}
+				elseif ($refundrequest == false)
+					$this->context->controller->errors[] = Tools::displayError('An error occured');
+				else
+					$this->context->controller->errors[] = Tools::displayError($refundrequest);
+			}
+		}
+	}
 
-	/* Check if the order was paid with Paylike and display the transaction details */
-	if (Db::getInstance()->getValue('SELECT module FROM '._DB_PREFIX_.'orders WHERE id_order = '.(int)Tools::getValue('id_order')) == $this->name)
+	public function hookDisplayAdminOrder($params)
 	{
+		$id_order = $params['id_order'];
+		$order = new Order((int)$id_order);
+		if ($order->module == $this->name)
+		{
+			$order_token = Tools::getAdminToken('AdminOrders'.(int)Tab::getIdFromClassName('AdminOrders').(int)$this->context->employee->id);
+			$this->context->smarty->assign(array(
+				'ps_version' => _PS_VERSION_,
+				'id_order' => $id_order,
+				'order_token' => $order_token
+			));
+			return $this->display(__FILE__, 'views/templates/hook/admin_order.tpl');
+		}
+	}
 
-		$order = new Order((int)Tools::getValue('id_order'));
-
-		$currency = $this->context->currency;
-		$c_char = $currency->sign;
-		$output = '
-		<script type="text/javascript">
-			$(document).ready(function() {
-				var appendEl;
-				if ($(\'select[name=id_order_state]\').is(":visible")) {
-					appendEl = $(\'select[name=id_order_state]\').parents(\'form\').after($(\'<div/>\'));
-				} else {
-					appendEl = $("#status");
+	public function hookActionOrderStatusPostUpdate($params)
+	{
+		$id_order			= (int)$params['id_order'];
+		$cart				= $params['cart'];
+		$order				= new Order((int)$id_order);
+		$order_state		= new OrderState(Tools::getValue('id_order_state'));
+		$order_currency 	= new Currency((int)$cart->id_currency);
+		$total				= Tools::ps_round($cart->getOrderTotal(true, Cart::BOTH), 2);
+
+		$total = Tools::convertPriceFull($total, $this->context->currency, $order_currency);
+		if ($order->module == $this->name && $order_state->shipped && Configuration::get('PAYLIKE_CHECKOUT_MODE') == 'delayed')
```

Comment:
As I was told earlier to capture delayed payment after order status has been marked as "shipped". Currently delayed payments are captured after order status is changed to "Shipped" by Admin.
These are the default order statuses that can be assigned to an order.

1. Awaiting check payment
2. Processing in progress
3. Shipped
4. Delivered
5. Canceled
6. Refund
7. Payment Error
8. On backoffice(paid)
9. Remote payment accepted
10. On backorder(not paid)

Please let me know when to capture the transaction.